arch/x86/kvm/pkvm/pkvm.c

```diff
@@ -565,6 +565,27 @@ static int pkvm_vcpu_put(int vm_handle, int vcpu_handle)
 	return ret;
 }
 
+static int pkvm_vcpu_handle_host_hypercall(struct kvm_vcpu *hvcpu, enum pkvm_hc hc,
+					   union pkvm_hc_data *out)
+{
+	struct kvm_vcpu *vcpu = this_cpu_read(cur_guest_vcpu);
+	int cpu = raw_smp_processor_id(), ret = 0;
+
+	if (!vcpu || hvcpu != this_cpu_read(host_vcpu))
```

Comment:
nit: `hvcpu != this_cpu_read(host_vcpu)` means a pKVM bug? If so, maybe put it in a WARN_ON_ONCE()?